test/infrastructure/docker/internal/loadbalancer/const.go

```diff
@@ -16,14 +16,16 @@ limitations under the License.
 
 package loadbalancer
 
-// Image defines the loadbalancer image name
-const Image = "haproxy"
+const (
```

Comment:
We could probably also change to use the "official" haproxy image, so

```go
Image = "haproxy-alpine"
DefaultImageRepository = "haproxytech"
DefaultImageTag        = "2.8"
```

Reply:
The issue (same as the issue for the Kind repo) is the official image doesn't have a config file attached which means it won't start. We could definitely workaround this by setting up the image in the code, but AFAIU CAPI needs to make some changes either way.

Reply:
We used the official image for the sample provider used in the kubecon na tutorial: https://github.com/capi-samples/cluster-api-provider-docker/tree/main/pkg/loadbalancer

I don't overly remember us having to change anything in the `loadbalancer` package but i can check.

Reply:
Let's please consider that the more we move away from what kind is doing (probably) the more maintenance effort we have

Reply:
Thanks for the pointer @richardcase - this is working. Looks like it's the 'global' haproxy image which doesn't have a config loaded. The haproxytech image has a default config which lets it run initially.

>Let's please consider that the more we move away from what kind is doing (probably) the more maintenance effort we have. After all CAPD clusters are still very closely modeled after kind clusters and there are dependencies.

I'm fine with pushing changing the image to a follow up - but I don't think there's additional maintenance effort here compared to using a third_party folder. The change is that this part of CAPD wasn't really maintained at all previously.

Reply:
Up to you. I didn't look into it in detail, but if you're sure that there are no dependencies to the kind haproxy I'm fine with switching to the "upstream" one. I just know that we have a lot of dependencies to the kindest/node image and even make sure that we start the node image exactly the same way as kind does.

If we don't have stuff like this for the lb image, I agree, it shouldn't matter

Reply:
The comment in the kind repo seems to match what Killian was writing above
> We cannot merely use the upstream haproxy image as haproxy will exit without a minimal config, so we introduce one that will list on the intended port and hot reload it at runtime with the actual desired config.
> https://github.com/kubernetes-sigs/kind/tree/main/images/haproxy

Reply:
Exactly - this image exists with the `_/haproxy` image, but not the `haproxytext/haproxy` image. 

I'm 100 percent fine with sticking with the haproxy image from kind for now btw - I don't see any motivation to change this. That said, making this directory first party will allow changes of that kind in future if someone is interested.

For now let's not change it.

Reply:
Sounds good!